addon/types/table.js

```diff
@@ -62,15 +60,12 @@ export default EmberObject.extend({
   init() {
     this._super(...arguments);
 
-    if (this.options?.features?.localStorage) {
-      if (!this.options.name) {
-        throw new Error(
-          '[Hypertable] Trying to use localStorage store without a store name. Please set "options.name".'
-        );
-      }
+    const originalColumnsChangeHook = this.hooks.onColumnsChange;
 
-      this.set('store', new LocalStorageStore(this.options.name));
-    }
+    this.hooks.onColumnsChange = () => {
+      this.options.store.updateState(this.columns);
+      originalColumnsChangeHook.apply(...arguments);
+    };
```

Comment:
*Why wrap it that way?*

The `onColumnsChange` hook is called at many places in the project but not only. Because we have specific projects defining their column/filters renderers, there are usages of `this.hooks.onColumnsChange` in crm-web and publishr-admin-web for example. That means that if we wanted to wrap it with a method from `Table`, it would mean go through all of the apps too.

Reply:
The main issue is maybe more why it's crm-web who "manually" call the `onColumnsChange` ? 
And what do you think about setup a pub/sub event system to be more flexible to attach action to some events ? (i know this involve changes :p)